wagtail_streamfield_migration_toolkit/test/tests/test_block_def_compare.py

```diff
@@ -34,170 +48,272 @@ class DiffThreeFieldStruct(StructBlock):
 
 class NestedStruct1(StructBlock):
     char_block = CharBlock()
-    struct_4f = FourFieldStruct()
-    struct_3f = ThreeFieldStruct()
+    nested_struct_one = FourFieldStruct()
+    nested_struct_two = ThreeFieldStruct()
 
 
 class NestedStruct2(StructBlock):
     char_block = CharBlock()
-    struct_new3f = ThreeFieldStruct()
-    struct_3f = ThreeFieldStruct()
+    nested_struct_one = ThreeFieldStruct()
+    nested_struct_two = ThreeFieldStruct()
 
 
-class NestedStruct3(StructBlock):
-    char_block = CharBlock()
-    struct_4f = FourFieldStruct()
-    struct_diff = DiffThreeFieldStruct()
+class BlockComparerTestCase(TestCase):
+    # TODO predicates
+    @staticmethod
+    def is_similar(score):
+        return score > CONFIDENT_SIMILARITY_THRESHOLD
+
+    @staticmethod
+    def is_unsure(score):
+        return (
+            score < CONFIDENT_SIMILARITY_THRESHOLD
+            and score > VERIFYING_SIMILARITY_THRESHOLD
+        )
 
+    @staticmethod
+    def is_dissimilar(score):
+        return score < VERIFYING_SIMILARITY_THRESHOLD
 
-class BlockComparerTestCase(TestCase):
     def assertBlockComparisonScore(
-        self, comparer, old_def, old_name, new_def, new_name, expected_score
+        self,
+        comparer: BaseBlockDefComparer,
+        old_def: Block,
+        old_name: str,
+        new_def: Block,
+        new_name: str,
+        predicate: Callable,
```

Comment:
If we're adding type annotations we can be more prescriptive about `predicate` - `Callable[[float], bool]`